### PR TITLE
address: handle missing types

### PR DESCRIPTION
**Pull request by danchr**
I ran into this on moradev.

### CODE REVIEW

File: mora/service/address.py [hunk 1]
```diff
@@ -333,12 +333,14 @@ def convert(start, end, effect):
...
 
-                yield {
-                    **get_one_address(c, addrrel, class_cache),
-                    keys.ADDRESS_TYPE: get_address_class(c, addrrel,
-                                                         class_cache),
-                    keys.VALIDITY: common.get_effect_validity(addrrel),
-                }
+                try:
+                    addr = get_one_address(c, addrrel, class_cache)
+                except KeyError as e:
```
Comment (cmoesgaard) [inline mora/service/address.py]: Where is KeyError thrown? Are we sure we're not swallowing actual exceptions here?

Comment (danchr) [reply mora/service/address.py]: > Where is KeyError thrown? Are we sure we're not swallowing actual exceptions here?

In [`get_one_class()`](https://github.com/magenta-aps/mora/pull/87/commits/6d113c1ac5bbf45ba28ef130fd13fcb3c2cf9874#diff-8426190c255ed9b2d05701905cef41c1R132) — we _could_ also return `None` there and handle that, but an exception seemed more appropriate for what ought to be an exceptional case…

File: mora/util.py [hunk 2]
```diff
@@ -289,3 +289,24 @@ def uniqueify(xs):
     '''return the contents of xs as a list, but stable'''
     # TODO: is this fast?
     return list(collections.OrderedDict(itertools.zip_longest(xs, ())).keys())
+
+
+def log_exception(msg='', with_traceback=False):
```
Comment (cmoesgaard) [inline mora/util.py]: `with_traceback` doesn't seem to be used.

File: mora/service/address.py (hunk 1 above)
Comment (cmoesgaard) [reply mora/service/address.py]: As far as I can tell from this, the UI is unaware that something went wrong. Should they be aware? Or do we just expect whoever is in charge to check the log ever so often for errors?

File: mora/service/address.py [hunk 3]
```diff
@@ -6,6 +6,10 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 
+import json
+import sys
+import traceback
```
Comment (cmoesgaard) [inline mora/service/address.py]: `sys` and `traceback` aren't used.

**Pull request merged**